### PR TITLE
Refactor config subpkg

### PR DESCRIPTION
**Pull request by leotrs**
## Motivation
The config system is working fine, but it is extremely difficult to debug and maintain. For example, I count *four* calls to `_run_config()`. This runs *the whole config dictionaries*. All of the logic to set the config is being ran at least four times. Furthermore, if you take a look at the config subpackage, you will find files such as `config/config_utils.py`, `config/config.py`, `config/logger.py`. The `config` dictionary is set in config.py, but the logic behind it is in config_utils. However, the logic behind logger is in logger.py, but it is set in config.py. This is hard to follow and debug.

## List of Changes
The config subpackage (i.e. everything living under `manim/config/`) has been refactored so that it does two things.
- First, *everything that is exposed to the user is set in `config/__init__.py`*. This sets the global `config`, `logger`, `console`, `file_writer_config`, etc. This is done by calling functions in the other files.

- Second, all the other files only contain utilities that serve to set, modify, inspect, or otherwise interact with those global things `config`, `logger`, `console`, `file_writer_config`, etc. For example, `config/utils.py` contains the logic behind setting `logger` and `config` for the whole library, for any use case, and `config/main_utils.py` contains the logic behind anything related to when manim is being called from the command line. In particular, it contains the logic behind parsing the CLI arguments.

## Benefits
- The config is no longer being set a whooping total of four times. Now it is only being set twice. Once when the library starts and once when the CLI arguments need to be parsed (and *only if* they need to be parsed).
- Much easier to debug and understand.
- In the future, anyone working on the config system only needs to change the utility files. As long as `config/__init__.py` exports the same global variables, the rest of the codebase doesn't need to know where or how those globals are being defined.

## Testing Status
All tests pass locally. However, *I think* I may have messed up the default logger output. If anyone could confirm and/or point to where I may have messed up, please do!

## Further Comments
- The only thing I don't like about this is that `tempconfig` had to be moved to `config/__init__.py`. This is unavoidable in the current model.
- I have not written documentation *because a lot of the code inside the utility files is going away soon*. Indeed, this PR is just one more in a series of PRs (#337). Once I'm done with the major idea, I will add documentation.
- The diff is large, but you will see that most of it is changing `import` statements to respect the new file organization. Most of the work here has been about just moving functions from one file to another.
- @eulertour there are currently no tests for the js renderer, could you please test that on your end?
- **The tests pass, but I would appreciate if people could pull this branch and render a few scenes on their machines before merging.**

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Comment by eulertour** [comment]
Regarding the JS renderer, you don't have to worry about breaking it right now. I'm going to overhaul it after the rendering logic is factored out of Scene.

**Review by eulertour — COMMENTED** [review]
Why switch these?

File: manim/__main__.py [hunk 1]
```diff
@@ -55,26 +54,38 @@ def open_file_if_needed(file_writer):
...
                     cfg_subcmds.show()
-                elif subcommand == "export":
+                elif args.subcmd == "export":
                     cfg_subcmds.export(args.dir)
             else:
-                logger.error("No argument provided; Exiting...")
+                logger.error("No subcommand provided; Exiting...")
+
+        # elif args.cmd == "some_other_cmd":
+        #     something_else_here()
```
Comment (eulertour) [inline manim/__main__.py:74]: Did you mean to remove this?

File: manim/__main__.py [hunk 2]
```diff
@@ -8,17 +8,16 @@
 import importlib.util
 import types
 
-from . import constants, logger, console, file_writer_config
-from .config.config import camera_config, args
-from .config import cfg_subcmds
-from .utils.module_ops import (
+from manim import constants, logger, console, config, file_writer_config
+from manim import Scene
+from manim.utils.module_ops import (
```
Comment (eulertour) [inline manim/__main__.py:13]: Why switch these to absolute?

**Review by eulertour — APPROVED** [review]
I generally think everything looks fine here. I basically couldn't follow the configuration logic at all before, and now I can understand it somewhat, so I'd call that an improvement.

File: manim/config/logger.py [hunk 3]
```diff
@@ -1,120 +0,0 @@
...
-    )
-    global logger
-    rich_handler.setLevel(verbosity)
-    logger.addHandler(rich_handler)
-
-
-def set_file_logger(log_file_path):
-    file_handler = logging.FileHandler(log_file_path, mode="w")
-    file_handler.setFormatter(JSONFormatter())
-    global logger
```
Comment (huguesdevimeux) [inline manim/config/logger.py:110]: Question : where did this go ? I couldn't find it in `config/utils.py`

File: manim/__main__.py (hunk 1 above)
Comment (leotrs) [reply manim/__main__.py:74]: Kinda. This is a poor man's documentation saying "this is where you'd put other commands in the future".

File: manim/config/logger.py (hunk 3 above)
Comment (leotrs) [reply manim/config/logger.py:110]: Currently, the only time at which the file logger is set is during `__main__.py`, so I moved this to `main_utils.py` (and refactored it a bit).

File: manim/__main__.py (hunk 2 above)
Comment (leotrs) [reply manim/__main__.py:13]: In general, I like to think of the `__main__.py` module as a script that is external to the library that just uses the library functionality. If this is possible, then everything that `__main__` does can be reproduced _and tested_ from within the library itself without having to call the command line entrypoint.

It was already (almost) possible. I switched these imports to absolute to make totally sure that this is possible and continues to be possible in the future.

File: manim/config/logger.py (hunk 3 above)
Comment (huguesdevimeux) [reply manim/config/logger.py:110]: I didn't see this file, sorry for that. But then why is `JsonFormatter` in `config/utils.py`? It is only used for the logs. Maybe you should move it to main_utils as well. 

Or separate logger stuff, I don't know.

Comment (leotrs) [reply manim/config/logger.py:110]: Yeah I'm not happy with how things turned out in terms of file organization.

Basically, `utils.py` contains everything needed by `__init__.py` to define things during library startup. Then `main_utils.py` contains everything needed by `__main__.py` to render a scene. I think that works but may not be the most logical way of separating things. And I think I missed JsonFormatter lol :sweat_smile: 

Don't worry, as I mentioned above, this is only one in a series of refactors, and my next PR already deals with these files.

Comment (huguesdevimeux) [reply manim/config/logger.py:110]: No problem then :)

**Review by huguesdevimeux — APPROVED** [review]
SGTM
